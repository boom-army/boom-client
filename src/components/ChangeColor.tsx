import PaletteIcon from '@mui/icons-material/Palette';
import { Stack, Typography } from "@mui/material";
import { ThemeContext } from "../contexts/theme";
import { useState, useContext } from "react";

export const ChangeColor = ({ iconProps, stackProps }: any) => {
  const { theme, setTheme } = useContext(ThemeContext);

  const colors = [
    "#CA2055",
    "#1DA1F2",
    "#FFAD1F",
    "#794BC4",
    "#F45D22",
    "#17BF63",
  ];

  const [currentColor, setCurrentColor] = useState(colors[0]);

  const changeColor = () => {
    let newAccentColor;

    if (colors.indexOf(currentColor) === colors.length - 1) {
      newAccentColor = colors[0];
    } else {
      newAccentColor = colors[colors.indexOf(currentColor) + 1];
    }    

    setTheme({
      ...theme,
      accentColor: newAccentColor,
      pallette: {
        primary: {
          main: newAccentColor,
        }
      }
    });

    setCurrentColor(newAccentColor);
  };

  return (
      <Stack onClick={changeColor} direction="row" {...stackProps} sx={{cursor: "pointer"}}>
        <PaletteIcon sx={iconProps} />
        <Typography variant="body1">Color</Typography>
      </Stack>
  );
};
