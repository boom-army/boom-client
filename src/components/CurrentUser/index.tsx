import { Box, Grid, IconButton } from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import { useToggleDrawer } from "../../hooks";
import { WalletMultiButton } from "@solana/wallet-adapter-material-ui";

export const CurrentUser = () => {
  const toggleDrawer = useToggleDrawer();
  
  return (
    <>
      <Grid
        container
        direction="row"
        justifyContent="flex-start"
        alignItems="center"
      >
        <Box mr={1} display="flex" alignItems="center" >
        <WalletMultiButton />
          <IconButton
            color="primary"
            onClick={toggleDrawer(true)}
            sx={{ ml: 1 }}
          >
            <MenuIcon />
          </IconButton>
        </Box>
      </Grid>
    </>
  );
};
