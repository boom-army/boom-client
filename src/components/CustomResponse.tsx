import React from "react";
import { styled } from "@mui/material/styles";

const Wrapper = styled("div")({
  height: "70vh",
  fontSize: "1.1rem",
  textAlign: "justify",
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  justifyContent: "center",
  padding: "4rem",
  "@media screen and (max-width: 530px)": {
    fontSize: "1rem",
  },
});

export const CustomResponse = ({ text }: any) => (
  <Wrapper>
    <p>{text}</p>
  </Wrapper>
);
