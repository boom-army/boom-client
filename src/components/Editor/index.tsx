import { AutoFocusPlugin } from "@lexical/react/LexicalAutoFocusPlugin";
import { AutoScrollPlugin } from "@lexical/react/LexicalAutoScrollPlugin";
import { CheckListPlugin } from "@lexical/react/LexicalCheckListPlugin";
import { ClearEditorPlugin } from "@lexical/react/LexicalClearEditorPlugin";
import { ContentEditable } from "@lexical/react/LexicalContentEditable";
import { HashtagPlugin } from "@lexical/react/LexicalHashtagPlugin";
import { LexicalEditor } from "lexical";
import { LinkPlugin } from "@lexical/react/LexicalLinkPlugin";
import { ListPlugin } from "@lexical/react/LexicalListPlugin";
import { OnChangePlugin } from "@lexical/react/LexicalOnChangePlugin";
import { ReactNode, useEffect, useState } from "react";
import { useLexicalComposerContext } from "@lexical/react/LexicalComposerContext";
import { useRef } from "react";

import { RichTextPlugin } from "./plugins/RichTextPlugin";
import { EditorContext } from "./EditorContext";
import { TabFocusPlugin } from "./plugins/TabFocusPlugin";

interface IEditorProps {
  children?: ReactNode;
  hashtagsEnabled?: boolean;
  autoLinkEnabled?: boolean;
  emojisEnabled?: boolean;
  actionsEnabled?: boolean;
  placeholder?: string;
  listMaxIndent?: number;
  initialEditorState?: string;
  isReadOnly?: boolean;
  onChange?: (editorState: string, editorInstance?: LexicalEditor) => void;
}

export const TextEditor = ({
  children,
  hashtagsEnabled = false,
  initialEditorState,
  isReadOnly = false,
  onChange,
}: IEditorProps) => {
  const [editor] = useLexicalComposerContext();
  const [activeEditor, setActiveEditor] = useState(editor);

  const editorStateRef = useRef(null);
  const placeholderComponent = <div>Write something...</div>;
  const scrollRef = useRef(null);

  useEffect(() => {
    editor.setReadOnly(isReadOnly);
  }, []);

  return (
    <EditorContext.Provider
      value={{ initialEditor: editor, activeEditor, setActiveEditor }}
    >
      {children}
      <div className={`editor-container`} ref={scrollRef}>
        <AutoFocusPlugin />
        <ClearEditorPlugin />
        {hashtagsEnabled && <HashtagPlugin />}
        <AutoScrollPlugin scrollRef={scrollRef} />

        <>
          <RichTextPlugin
            contentEditable={<ContentEditable />}
            placeholder={placeholderComponent}
            initialEditorState={initialEditorState}
          />
          <OnChangePlugin
            ignoreInitialChange={false}
            onChange={(editorState) => {
              onChange?.(JSON.stringify(editorState), activeEditor);
              return (editorStateRef.current = editorState);
            }}
          />
          <ListPlugin />
          <CheckListPlugin />
          <LinkPlugin />
          <TabFocusPlugin />
        </>
      </div>
    </EditorContext.Provider>
  );
};
