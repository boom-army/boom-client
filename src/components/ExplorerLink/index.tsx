import React from "react";
import { Typography } from "@mui/material";
import { shortenAddress } from "../../utils/utils";
import { PublicKey } from "@solana/web3.js";

export const ExplorerLink = (props: {
  address: string | PublicKey;
  type: string;
  code?: boolean;
  style?: React.CSSProperties;
  length?: number;
}) => {
  const { type, code } = props;

  const address =
    typeof props.address === "string"
      ? props.address
      : props.address?.toBase58();

  if (!address) {
    return null;
  }

  const length = props.length ?? 4;

  return (
    <a
      href={`https://explorer.solana.com/${type}/${address}`}
      target="_blank"
      title={address}
      style={props.style}
      rel="noreferrer"
    >
      {code ? (
        <Typography>{shortenAddress(address, length)}</Typography>
      ) : (
        shortenAddress(address, length)
      )}
    </a>
  );
};
