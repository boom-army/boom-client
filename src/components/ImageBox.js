import { SRLWrapper } from "simple-react-lightbox";
import { styled } from "@mui/material/styles";

const ImageBoxWrapper = styled("div")((props) => ({
  "> div": {
    display: "flex",
    borderRadius: "16px",
    border: `1px solid ${props.theme.secondaryColor}`,
    overflow: "hidden",
    width: "90%",
    flexWrap: "wrap",
    margin: "0 0 0.75rem",
  },

  "> div > .tweet-image-col": {
    display: "flex",
    flex: "50%",
    width: "100%",
  },
}));

const ImageColumn = styled("div")((props) => ({
  height: "100%",
  display: "flex",
  flexDirection: "column",
  flex: "50%",
  marginRight: props.fileslength > 1 ? "2px;" : "0;",
}));

const ImageWrapper = styled("div")((props) => ({
  lineHeight: "0",
  height: "100%",
  marginBottom: props.hasbottommargin ? "2px;" : "0;",
}));

const Image = styled("img")((props) => ({
  objectFit: "cover",
  maxWidth: "100%",
  width: "100%",
  height: "100%",
  position: "relative",
  cursor: props.disablelightbox ? "inherit" : "pointer",
}));

const ImageColumnContainer = ({ files, isLeftCol, disablelightbox }) => {
  const className = `tweet-image-col ${
    isLeftCol ? "tweet-image-left-col" : "tweet-image-right-col"
  }`;
  return (
    <div className={className}>
      <ImageColumn fileslength={files.length}>
        {files.map((file, index, arr) => (
          <ImageWrapper
            key={file.id}
            hasfullheight={arr.length === 1 ? "true" : "false"}
            hasbottommargin={arr.length > 1 && index === 0 ? "true" : "false"}
            fileindex={index}
          >
            <Image
              disablelightbox={disablelightbox}
              key={file.id}
              src={file.url}
              alt="tweet-file"
            />
          </ImageWrapper>
        ))}
      </ImageColumn>
    </div>
  );
};

export const ImageBox = ({ files, disablelightbox }) => {
  const options = {
    settings: {
      lightboxTransitionSpeed: 0,
    },
    buttons: {
      showAutoplayButton: false,
      showDownloadButton: false,
      showThumbnailsButton: false,
    },
  };

  const leftColumnFiles = files.filter((_, index, arr) => {
    return index === 0 || (arr.length === 4 && index === 2);
  });

  const rightColumnFiles = files.filter((_, index, arr) => {
    return (
      index === 1 ||
      (arr.length === 3 && index === 2) ||
      (arr.length === 4 && index === 3)
    );
  });

  if (disablelightbox)
    return (
      <ImageBoxWrapper>
        {!!files.length && (
          <div>
            {!!leftColumnFiles.length && (
              <ImageColumnContainer
                disablelightbox
                files={leftColumnFiles}
                isLeftCol={true}
              />
            )}
            {!!rightColumnFiles.length && (
              <ImageColumnContainer
                disablelightbox
                files={rightColumnFiles}
                isLeftCol={false}
              />
            )}
          </div>
        )}
      </ImageBoxWrapper>
    );

  return (
    <ImageBoxWrapper>
      {!!files.length && (
        <SRLWrapper options={options}>
          {!!leftColumnFiles.length && (
            <ImageColumnContainer files={leftColumnFiles} isLeftCol={true} />
          )}
          {!!rightColumnFiles.length && (
            <ImageColumnContainer files={rightColumnFiles} isLeftCol={false} />
          )}
        </SRLWrapper>
      )}
    </ImageBoxWrapper>
  );
};
