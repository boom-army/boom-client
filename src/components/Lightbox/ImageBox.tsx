import { SRLWrapper } from "simple-react-lightbox";
import { styled } from "@mui/material/styles";

const ImageBoxWrapper = styled("div")((props: any) => ({
  "> div": {
    display: "flex",
    borderRadius: "0.2em",
    border: `1px solid ${props.theme.secondaryColor}`,
    overflow: "hidden",
    maxWidth: "400px",
    width: "100%",
    flexWrap: "wrap",
    margin: "0.75rem 0",
  },

  "> div > .tweet-image-col": {
    display: "flex",
    flex: "50%",
    width: "100%",
  },
}));

const ImageColumn = styled("div")<{ fileslength?: any; props?: any }>(
  (props) => ({
    height: "100%",
    display: "flex",
    flexDirection: "column",
    flex: "50%",
    marginRight: props.fileslength > 1 ? "2px;" : "0;",
  })
);

const ImageWrapper = styled("div")<{
  hasfullheight?: any;
  hasbottommargin?: any;
  fileindex?: any;
  props?: any;
}>({
  lineHeight: "0",
  height: "100%",
});

const Image = styled("img")<{ disablelightbox?: any; props?: any }>(
  (props) => ({
    objectFit: "cover",
    maxWidth: "400px",
    width: "100%",
    height: "100%",
    position: "relative",
    cursor: props.disablelightbox ? "inherit" : "pointer",
  })
);

interface Props {
  files?: any;
  isLeftCol?: boolean;
  disablelightbox?: any;
}

const ImageColumnContainer = ({ files, isLeftCol, disablelightbox }: Props) => {
  const className = `tweet-image-col ${
    isLeftCol ? "tweet-image-left-col" : "tweet-image-right-col"
  }`;  

  return (
    <div className={className}>
      <ImageColumn fileslength={files.length}>
        {files.map((file: any, index: number, arr: any) => (
          <ImageWrapper
            key={file.id}
            hasfullheight={arr.length === 1 ? "true" : "false"}
            hasbottommargin={arr.length > 1 && index === 0 ? "true" : "false"}
            fileindex={index}
          >
            <Image
              disablelightbox={disablelightbox}
              key={file.id}
              src={file.url}
              alt="tweet-file"
            />
          </ImageWrapper>
        ))}
      </ImageColumn>
    </div>
  );
};

export const ImageBox = ({ files, disablelightbox }: Props) => {
  const options = {
    settings: {
      lightboxTransitionSpeed: 0,
    },
    buttons: {
      showAutoplayButton: false,
      showDownloadButton: false,
      showThumbnailsButton: false,
    },
  };

  const leftColumnFiles = files.filter((_: any, index: number, arr: any) => {
    return index === 0 || (arr.length === 4 && index === 2);
  });

  const rightColumnFiles = files.filter((_: any, index: number, arr: any) => {
    return (
      index === 1 ||
      (arr.length === 3 && index === 2) ||
      (arr.length === 4 && index === 3)
    );
  });

  if (disablelightbox)
    return (
      <ImageBoxWrapper>
        {!!files.length && (
          <div>
            {!!leftColumnFiles.length && (
              <ImageColumnContainer
                disablelightbox
                files={leftColumnFiles}
                isLeftCol={true}
              />
            )}
            {!!rightColumnFiles.length && (
              <ImageColumnContainer
                disablelightbox
                files={rightColumnFiles}
                isLeftCol={false}
              />
            )}
          </div>
        )}
      </ImageBoxWrapper>
    );

  return (
    <ImageBoxWrapper>
      {!!files.length && (
        <SRLWrapper options={options}>
          {!!leftColumnFiles.length && (
            <ImageColumnContainer files={leftColumnFiles} isLeftCol={true} />
          )}
          {!!rightColumnFiles.length && (
            <ImageColumnContainer files={rightColumnFiles} isLeftCol={false} />
          )}
        </SRLWrapper>
      )}
    </ImageBoxWrapper>
  );
};
