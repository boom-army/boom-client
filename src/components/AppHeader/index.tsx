import { useCallback, useEffect, useMemo, useContext } from "react";
import base58 from "bs58";
import {
  AppBar,
  Container,
  Grid,
  Toolbar,
} from "@mui/material";
import { CurrentUser } from "../CurrentUser";
import { PUBLIC_ADDRESS, LOGIN_REGISTER } from "../../queries/auth";
import { USER_FOLLOW } from "../../queries/follow";
import { Theme } from "../../contexts/theme";
import { UserContext } from "../../contexts/user";
import { useMutation } from "@apollo/client";
import { useSnackbar } from "../../contexts/snackbar";
import { useWallet } from "@solana/wallet-adapter-react";
import { FeedDocument } from "../../generated/graphql";
import { Box } from "@mui/system";
import { Link } from "react-router-dom";
import { RoutePath } from "../../constants";

export const AppHeader = () => {
  const { connected, wallet, publicKey, signMessage } = useWallet();
  const { enqueueSnackbar } = useSnackbar();
  const { setUser } = useContext(UserContext);

  const token = localStorage.getItem("token");
  const headerImg =
    localStorage.getItem(Theme.StorageTag) === Theme.Light
      ? "/assets/boom-logo-dark.png"
      : "/assets/boom-logo-light.png";

  const [getNonce] = useMutation(PUBLIC_ADDRESS);
  const [setLogin] = useMutation(LOGIN_REGISTER, {
    refetchQueries: [{ query: FeedDocument }, { query: USER_FOLLOW }],
    onCompleted({ loginRegister }) {
      if (localStorage) {
        setUser(loginRegister.user);
        localStorage.setItem("token", loginRegister.token);
      }
    },
  });

  const walletPublicKey = useMemo(() => publicKey?.toBase58(), [publicKey]);

  const signin = useCallback(async () => {
    if (!walletPublicKey) {
      enqueueSnackbar("Wallet not connected!", { variant: "error" });
      return;
    }

    if (!signMessage) {
      enqueueSnackbar("Wallet does not support message signing!", {
        variant: "error",
      });
      return;
    }

    try {
      const {
        data: { address },
      } = await getNonce({
        variables: { publicAddress: walletPublicKey },
      });
      if (address.hasPublicAddress) {
        const data = new TextEncoder().encode(address.user.nonce);
        const signature = await signMessage(data);

        await setLogin({
          variables: {
            publicAddress: walletPublicKey,
            signature: base58.encode(signature),
          },
        });
        window.location.reload();
        enqueueSnackbar(
          `Wallet ${walletPublicKey} connected to account. Happy posting.`,
          { variant: "success" }
        );
      } else {
        await setLogin({ variables: { publicAddress: walletPublicKey } });
        enqueueSnackbar(`Wallet ${walletPublicKey} created for account.`, {
          variant: "success",
        });
      }
    } catch (error) {
      enqueueSnackbar(`Error connecting: ${error}`, { variant: "error" });
    }
  }, [signMessage, getNonce, setLogin, walletPublicKey, enqueueSnackbar]);

  useEffect(() => {
    if (wallet && !token && connected) signin();
  }, [wallet, signin, token, connected]);

  // const logout = () => {
  //   localStorage.clear();
  //   window.location.replace('/');
  // };

  return (
    <AppBar
      position="static"
      sx={{
        borderBottom: (t) => `1px solid ${t.palette.divider}`,
      }}
    >
      <Toolbar variant="dense">
        {/* Remove this when we go full width */}
        <Container maxWidth="lg">
          <Grid container>
            <Grid item xs={12}>
              <Box display="flex" justifyContent="space-between">
                <Box mr={1} display="flex" alignItems="center">
                  <Link
                    to={RoutePath.HOME}
                    style={{ display: "flex", alignItems: "center" }}
                  >
                    <img src={headerImg} alt="Boom" height={20} />
                  </Link>
                </Box>
                <Box mt={0.5}>
                  <CurrentUser />
                </Box>
              </Box>
            </Grid>
          </Grid>
        </Container>
      </Toolbar>
    </AppBar>
  );
};
