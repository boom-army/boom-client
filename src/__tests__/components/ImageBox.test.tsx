import React from "react";
import { ImageBox } from "../../components/ImageBox";
import { act, fireEvent, render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { createSerializer } from '@emotion/jest'

expect.addSnapshotSerializer(createSerializer(
  {
    classNameReplacer(className, index) {
      return `css-${index}`
    },
     // includeStyles: false 
  }
));


describe("<ImageBox/> component :", () => {
  test("display <ImageBox/> component ", async () => {
    let rendered = render(<ImageBox files={[]} />, { wrapper: MemoryRouter });
    expect(rendered).toMatchSnapshot();
  });
});
