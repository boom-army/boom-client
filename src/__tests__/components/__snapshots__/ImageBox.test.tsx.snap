// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageBox/> component : display <ImageBox/> component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .css-0>div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 16px;
  border: 1px solid undefined;
  overflow: hidden;
  width: 90%;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 0 0.75rem;
}

.css-0>div>.tweet-image-col {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 50%;
  -ms-flex: 50%;
  flex: 50%;
  width: 100%;
}

<body>
    <div>
      <div
        class="css-0"
      />
    </div>
  </body>,
  "container": .css-0>div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 16px;
  border: 1px solid undefined;
  overflow: hidden;
  width: 90%;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 0 0.75rem;
}

.css-0>div>.tweet-image-col {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 50%;
  -ms-flex: 50%;
  flex: 50%;
  width: 100%;
}

<div>
    <div
      class="css-0"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
